notify/telegram/telegram.go

```diff
@@ -44,7 +47,19 @@ func New(conf *config.TelegramConfig, t *template.Template, l log.Logger, httpOp
 		return nil, err
 	}
 
-	client, err := createTelegramClient(conf.BotToken, conf.APIUrl.String(), conf.ParseMode, httpclient)
+	var u string
+	if conf.BotToken != "" {
+		u = string(conf.BotToken)
+	}
+	if len(conf.BotTokenFile) > 0 {
+		content, err := os.ReadFile(conf.BotTokenFile)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to read bot token from file")
+		}
+		u = strings.TrimSpace(string(content))
+	}
+
+	client, err := createTelegramClient(u, conf.APIUrl.String(), conf.ParseMode, httpclient)
```

Comment:
I think that the telegram client should be created in the Notify() function otherwise if the token file is updated, Alertmanager wouldn't pick it up.